cli/bench_command.go

```diff
@@ -183,66 +220,96 @@ func (c *benchCmd) bench(_ *kingpin.ParseContext) error {
 				}
+			} else if (c.pull || c.pushDurable) && c.numSubs > 0 {
+				log.Printf("Using stream: %s", c.streamName)
 			}
-		}()
 
-		if c.streamName == DEFAULT_STREAM_NAME {
-			// create the stream with our attributes, will create it if it doesn't exist or make sure the existing one has the same attributes
-			_, err = js.AddStream(&nats.StreamConfig{Name: c.streamName, Subjects: []string{c.subject}, Retention: nats.LimitsPolicy, Storage: storageType, Replicas: c.replicas})
-			if err != nil {
-				log.Fatalf("there is already a stream %s defined with conflicting attributes, if you want to delete and re-define the stream use `nats stream delete` (%v)", c.streamName, err)
+			if c.purge {
+				log.Printf("Purging the stream")
+				err = js.PurgeStream(c.streamName)
+				if err != nil {
+					log.Fatalf("error purging stream %s: %v", c.streamName, err)
+				}
 			}
-		} else if c.js && c.pull && c.numSubs > 0 {
-			log.Printf("Using stream: %s", c.streamName)
-		}
 
-		if c.purge {
-			log.Printf("Purging the stream")
-			err = js.PurgeStream(c.streamName)
-			if err != nil {
-				log.Fatalf("error purging stream %s: %v", c.streamName, err)
+			// create the pull consumer
+			if c.numSubs > 0 {
+				if c.pull {
+					_, err = js.AddConsumer(c.streamName, &nats.ConsumerConfig{
+						Durable:       c.consumerName,
+						DeliverPolicy: nats.DeliverAllPolicy,
+						AckPolicy:     nats.AckExplicitPolicy,
+						ReplayPolicy:  nats.ReplayInstantPolicy,
+						MaxAckPending: func(a int) int {
+							if a >= 20000 {
+								return a
+							} else {
+								return 20000
+							}
+						}(c.numSubs * c.consumerBatch),
+					})
+					if err != nil {
+						log.Fatal("error creating the pull consumer: ", err)
+					}
+					defer func() {
+						err := js.DeleteConsumer(c.streamName, c.consumerName)
+						if err != nil {
+							log.Printf("error deleting the pull consumer on stream %s: %v", c.streamName, err)
+						}
+						log.Printf("Deleted durable consumer: %s\n", c.consumerName)
+
+					}()
+					log.Printf("Defined durable explicitly acked pull consumer: %s\n", c.consumerName)
+				} else if c.pushDurable {
+					_, err = js.AddConsumer(c.streamName, &nats.ConsumerConfig{
+						Durable:        c.consumerName,
+						DeliverSubject: c.consumerName + "-DELIVERY",
+						DeliverGroup:   c.consumerName + "-GROUP",
+						DeliverPolicy:  nats.DeliverAllPolicy,
+						AckPolicy:      nats.AckExplicitPolicy,
+						ReplayPolicy:   nats.ReplayInstantPolicy,
+						MaxAckPending:  c.consumerBatch * c.numSubs,
+					})
+					if err != nil {
+						log.Fatal("error creating the durable push consumer: ", err)
+					}
+					defer func() {
+						err := js.DeleteConsumer(c.streamName, c.consumerName)
+						if err != nil {
+							log.Printf("error deleting the durable push consumer on stream %s: %v", c.streamName, err)
```

Comment:
should be fatal?

Reply:
Sure, changed it (though it doesn't make much difference since this is cleanup code)